tests/trestle/utils/list_utils_test.py

```diff
@@ -81,3 +84,39 @@ def test_get_item_from_list() -> None:
     item = list_utils.get_item_from_list(items, 4, lambda x: list(x.values())[0])
     assert item == {'b': 4}
     assert items == [{'a': 2}, {'b': 4}, {'c': 6}]
+
+
+def test_comma_sep_to_list() -> None:
+    """Test comma sep to list."""
+    assert list_utils.comma_sep_to_list(None) == []
+    assert list_utils.comma_sep_to_list('a') == ['a']
+    assert list_utils.comma_sep_to_list('') == ['']
+    assert list_utils.comma_sep_to_list(' ab , c d,  ef  ') == ['ab', 'c d', 'ef']
+
+
+def test_comma_colon_sep_to_dict() -> None:
+    """Test comma colon sep to dict."""
+    assert list_utils.comma_colon_sep_to_dict(None) == {}
+    assert list_utils.comma_colon_sep_to_dict('a') == {'a': 'a'}
+    assert list_utils.comma_colon_sep_to_dict('') == {'': ''}
+    assert list_utils.comma_colon_sep_to_dict('  ') == {'': ''}
+    assert list_utils.comma_colon_sep_to_dict(' ab: cd ,de:fg, h , i : j k, m: n : op ') == {
```

Comment:
I would expect the last case i.e. `m: n: op` to be `{m: { n: op }}` ?

Reply:
The intent is for a single colon to create a mapping from one string to another.  Doing it as shown in the tests allows a mapping to a string that contains colons - so it is working as designed.

Reply:
Okay, this is specific for `--sections` use case then

Reply:
since commas are not allowed in the second string and only colons - it would only allow a deep tree with no branches - so it would have little value.  Yes these are for sections parsing, but I think there is general use for 'str1:str2,str3:str4' simple string mapping